substrate/frame/identity/src/lib.rs

```diff
@@ -1141,17 +1188,37 @@ pub mod pallet {
 		/// accepted by the user and must now be beyond its expiration. The call must include the
 		/// full username, as in `username.suffix`.
 		#[pallet::call_index(19)]
-		#[pallet::weight(T::WeightInfo::remove_expired_approval())]
+		#[pallet::weight(T::WeightInfo::remove_expired_approval(0))]
 		pub fn remove_expired_approval(
 			origin: OriginFor<T>,
 			username: Username<T>,
 		) -> DispatchResultWithPostInfo {
 			let _ = ensure_signed(origin)?;
-			if let Some((who, expiration)) = PendingUsernames::<T>::take(&username) {
+			if let Some((who, expiration, provider)) = PendingUsernames::<T>::take(&username) {
 				let now = frame_system::Pallet::<T>::block_number();
 				ensure!(now > expiration, Error::<T>::NotExpired);
+				let actual_weight = match provider {
+					Provider::AuthorityDeposit(deposit) => {
+						let suffix = Self::suffix_of_username(&username)
+							.ok_or(Error::<T>::InvalidUsername)?;
+						let authority_account = AuthorityOf::<T>::get(&suffix)
+							.map(|auth_info| auth_info.account_id)
+							.ok_or(Error::<T>::NotUsernameAuthority)?;
+						let err_amount = T::Currency::unreserve(&authority_account, deposit);
+						debug_assert!(err_amount.is_zero());
+						T::WeightInfo::remove_expired_approval(0)
+					},
+					Provider::Allocation => {
+						// We don't refund the allocation, it is lost, but we refund some weight.
+						T::WeightInfo::remove_expired_approval(1)
```

Comment:
maybe having some const or enum instead of 0 and 1 would make it more readable.

But it is good to me.

Reply:
Could just pass the Provider Enum? Similarly in other places.